File: backend/app/schemas/analytics.py
from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field


class NoShowRiskScore(BaseModel):
    appointment_id: str
    risk_score: float = Field(ge=0, le=1)
    risk_tier: str
    top_factors: list[str] = Field(default_factory=list)


class DemandForecastResponse(BaseModel):
    specialty: str
    date: date
    expected_demand: int = Field(ge=0)
    confidence_interval: tuple[int, int]

File: frontend/src/api/analytics.ts
import { apiClient } from "./client";

export interface NoShowScore {
  appointment_id: string;
  risk_score: number;
  risk_tier: string;
  top_factors: string[];
}

export interface DemandForecast {
  specialty: string;
  date: string;
  expected_demand: number;
  confidence_interval: [number, number];
}

export const fetchNoShowScore = async (appointmentId: string) => {
  const { data } = await apiClient.get<NoShowScore>(`/analytics/no-show/${appointmentId}`);
  return data;
};

export const fetchDemandForecast = async (specialty: string, horizonDays = 7) => {
  const { data } = await apiClient.get<DemandForecast[]>("/analytics/demand", {
    params: { specialty, horizon_days: horizonDays },
  });
  return data;
};
